mobsf/static/others/js/suppression.js

```diff
@@ -0,0 +1,153 @@
+    } else {
+      endpoint = '/suppress_by_rule/'
+      title = '<strong>Disable Rule</strong>'
+      html = `This will suppress the rule <b>${escapeHtml(rule)}</b> from tiggering for <b>${escapeHtml(pkg)}</b> from now on.`
+    }
+    if (manifest){
+      table = '#table_manifest'
+      type = 'manifest'
+    } else {
+      table = '#table_code'
+      type = 'code'
+    }
+
+    Swal.fire({
+      title: title,
+      type: 'warning',
+      html: html,
+      showCancelButton: true,
+      cancelButtonText: 'Cancel',
+      confirmButtonText: 'Suppress',
+    }).then((result) => {
+      if (result.value) {
+        action(document.location.origin + endpoint,  { checksum: hash, rule, type }, function(json) {
+            if (json.status==="ok") {
+                $(table).DataTable().row(tr).remove().draw();
+            } else {
+              Swal.fire("Failed to Suppress")
+            }
+        });
+      }
+    });
+      
+  }
+
+function remove_suppression(ctx){
+  kind = $(ctx).data('kind');
+  rule = $(ctx).data('rule');
+  type = $(ctx).data('type');
+  
+  Swal.fire({
+    title: 'Delete suppression rule?',
+    type: 'warning',
+    text: 'Do you want to delete the suppression rule?',
+    showCancelButton: true,
+    cancelButtonText: 'Cancel',
+    confirmButtonText: 'Delete',
+  }).then((result) => {
+    if (result.value) {
+      action(document.location.origin + '/delete_suppression/', { checksum: hash, rule, kind, type }, function(json) {
+          if (json.status==="ok") {
+            window.location.hash = 'suppression';
+            window.location.reload();
+          } else {
+            Swal.fire("Failed to remove suppression rule")
+          }
+      });
+    }
+  });
+
+}
+
+function get_rules(type, rules){
+  var html = ''
+  rules.forEach(element => {
+    html += `${escapeHtml(element)} - <a onclick='remove_suppression(this)' data-rule='${escapeHtml(element)}' data-type='${type}' data-kind='rule'><i class="fa fa-trash fa-2xs"></i></a></br>`
+  });
+  return html
+}
+
+
+function get_files(type, files){
+  var html = ''
+  for (const [rule, rfiles] of Object.entries(files)) {
+    html += `<b>${escapeHtml(rule)}</b> - <a onclick='remove_suppression(this)' data-rule='${escapeHtml(rule)}' data-type='${type}' data-kind='file'><i class="fa fa-trash fa-2xs"></i></a></br>`
+    html += `<a class="btn btn-primary btn-sm" data-toggle="collapse" href="#c_${slugify(escapeHtml(rule))}" role="button" aria-expanded="false" aria-controls="c_${slugify(escapeHtml(rule))}">Files ➜</a><div class="collapse" id="c_${slugify(escapeHtml(rule))}"><div class="card card-body">`
+    rfiles.forEach(element => {
+      html += `<li>${escapeHtml(element)}</li>`
+    });
+    html += '</div></div></br>'
+  }
+  return html
+}
+
+function list_suppressions(){
+  action(document.location.origin + '/list_suppressions/', { checksum: hash }, function(json) {
```

Comment:
Don't you prefer using url names? this is prone to some errors I guess

Reply:
I can update that.